src/Pos/Block/Logic/Header.hs

```diff
@@ -70,13 +70,13 @@ mkCHRinvalid = CHInvalid . T.intercalate "; "
 -- | Classify new header announced by some node. Result is represented
 -- as ClassifyHeaderRes type.
 classifyNewHeader
-    :: (MonadSlots m, SscHelpersClass ssc, MonadDBPure m)
+    :: forall m ssc. (MonadSlots m, DB.MonadBlockDB ssc m)
```

Comment:
`MonadBlockDB` is more severe constraint than `MonadDBPure`, but probably we can live with it.